File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_094726/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

interface UserRecord {
  Name: string;
  Age: number;
  Email: string;
  Score: number;
}

function solve() {
  try {
    // Path to the input file
    const filePath = path.join('input', 'data.csv');
    
    // Read the file content as a string
    const content = fs.readFileSync(filePath, 'utf8');
    
    // Split by newline and filter out any empty lines (e.g., at the end of the file)
    const lines = content.split(/\r?\n/).filter(line => line.trim().length > 0);
    
    if (lines.length === 0) {
      console.log(JSON.stringify([]));
      return;
    }

    // Remove the header row
    const dataLines = lines.slice(1);
    
    const results: UserRecord[] = dataLines.map(line => {
      const columns = line.split(',');
      
      return {
        Name: columns[0]?.trim() || '',
        Age: parseInt(columns[1]?.trim() || '0', 10),
        Email: columns[2]?.trim() || '',
        Score: parseFloat(columns[3]?.trim() || '0')
      };
    });

    // Output the final array as JSON
    process.stdout.write(JSON.stringify(results, null, 2) + '\n');
  } catch (error) {
    // In a real scenario, we might handle errors differently, 
    // but here we'll just exit if the file cannot be read.
    process.exit(1);
  }
}

solve();

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_094726/python/001_csv_to_json/solution.py
import csv
import json
import sys

def main():
    input_file = 'input/data.csv'
    data = []

    try:
        with open(input_file, mode='r', encoding='utf-8') as csvfile:
            reader = csv.DictReader(csvfile)
            for row in reader:
                # Construct the object with specified type conversions
                item = {
                    "Name": row['Name'],
                    "Age": int(row['Age']),
                    "Email": row['Email'],
                    "Score": float(row['Score'])
                }
                data.append(item)
        
        # Output the resulting list as a JSON array to stdout
        sys.stdout.write(json.dumps(data))
        
    except FileNotFoundError:
        # If the file is missing, we output an empty array to remain valid JSON
        sys.stdout.write(json.dumps([]))
    except Exception:
        # Generic handler to ensure we don't output non-JSON text on crash
        sys.exit(1)

if __name__ == "__main__":
    main()